core/src/main/java/org/infinispan/protostream/ProtobufUtil.java

```diff
@@ -70,18 +69,22 @@ private static <A> void write(ImmutableSerializationContext ctx, TagWriterImpl o
       out.flush();
    }
 
+   public static void writeTo(ImmutableSerializationContext ctx, RandomAccessOutputStream out, Object t) throws IOException {
+      write(ctx, TagWriterImpl.newInstance(ctx, out), t);
+   }
+
    public static void writeTo(ImmutableSerializationContext ctx, OutputStream out, Object t) throws IOException {
       write(ctx, TagWriterImpl.newInstance(ctx, out), t);
```

Comment:
nitpicking, you probably want to do `out instanceof RandomAccessOutputStream` as you did in `toWrappedStream` method.